packages/site/src/polyfills.js

```diff
@@ -1 +1,2 @@
-import 'core-js/fn/object/assign'
+import 'core-js/es6'
+import 'core-js/es7'
```

Comment:
es7 has `Object.values` as well as `Array.includes` which we use

Reply:
I like the ease of this., but I'm still concerned that it hides the issue from us a bit and from our users.

From us because we "deploy the DS to an app, the site, and the components work".  We think, "well, the next BC that imports these components will have it work for them as well."

From our users because they come to the site, see it working in IE, go import it in their site, it *may* not work, and they wonder what's up.

To expose this a bit more to ourselves and our users, I think we need to either need to: 

- Track what's polyfilled on the site (our only IE11, team-owned runtime) and document in the dev docs so users can make sure they polyfill one way or the other. // pro: if they have a mega polyfill, they're covered, if they're trying to play it light, they'll see the features that they need polyfilled and grab those as one-offs.
- Or build the polyfills into the components that need them for the IE11 (and other) runtimes that we support. // pro: users of our components don't have to worry, it's taken care of;  con: potential bloat from multiple copies of the same polyfill (no tree shaking without es modules)

What are your thoughts on this?  Maybe something to CPT?  Or maybe add this to a Dash search term (core-js/babel-polyfill)?

Reply:
yep, i opened this as a conversation starter. and I have the same concerns and have the same solutions you have outlined.

I think up until this point we've probably been assuming that consumers have something like `babel-polyfill` and that has gotten us this far. adding that expectation to integration docs is definitely something we could do.

we have bundled in other polyfills(resize-observer, focus-within) but I think that's because those are outside of what `core-js`/`babel` provide. would including `babel-polyfill` in our bundle cause large file sizes? could it get bundled twice?

I lean more towards making `babel-polyfill` a requirement by the consuming dev team. they can include the entire polyfill(probably pretty common) or setup webpack/babel `preset-env` to include only the necessary polyfills.

Reply:
Yes, I too believe it's the general use of babel-polyfill/equivalent in projects that has probably allowed us to get this far w/o hearing about it much.  Given that good piece of likely evidence, explicitly requiring users to use a polyfill (via docs) seems reasonable.

Yes, including the whole babel-polyfill will probably make our components unreasonably large.  That said, I haven't attempted it.  We could test that expectation.  And yes, it could definitely get bundled twice (or MORE! -- `n + 1` DS  omponents + their own BC) because we lack an es module build.

We already use `babel-preset-env`, of course.  Babel is the transpiling, converting our new age syntax to browser-ready syntax.  But separate polyfills per browser-absent *features* are still required.  Babel does not add those, or at least preset-env does not.